src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.test.ts

```diff
@@ -3,10 +3,11 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { of } from 'rxjs';
 import { SavedObjectsErrorHelpers } from '../../../../core/server';
 import { WorkspaceSavedObjectsClientWrapper } from './workspace_saved_objects_client_wrapper';
 
-const generateWorkspaceSavedObjectsClientWrapper = () => {
+const generateWorkspaceSavedObjectsClientWrapper = (isDashboardAdmin = false) => {
```

Comment:
Can we add an override argument `getPrincipalsFromRequest` or using `permissionControlMock. getPrincipalsFromRequest.mockReturnValue` instead? Add a isDashboardAdmin may not generic.

Reply:
Using `permissionControlMock. getPrincipalsFromRequest.mockReturnValue`  is invalid. I had add a `role` as a argument argument.